catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/geofeature/GazetteerFeatureService.java

```diff
@@ -56,35 +75,82 @@ public void setGeoEntryQueryable(GeoEntryQueryable geoEntryQueryable) {
   }
 
   @Override
-  public Feature getFeatureByName(String name) {
+  public SimpleFeature getFeatureByName(String name) {
+    List<GeoEntry> entries;
     try {
-      List<GeoEntry> entries = this.geoEntryQueryable.query(name, 1);
-      if (!entries.isEmpty()) {
-        GeoResult geoResult = getGeoResultFromGeoEntry(entries.get(0));
-        return getFeatureFromGeoResult(geoResult);
-      }
+      entries = this.geoEntryQueryable.query(name, 1);
     } catch (GeoEntryQueryException e) {
       LOGGER.debug("Error while making feature service request.", e);
+      return null;
+    }
+    if (entries.isEmpty()) {
+      return null;
+    }
+    GeoEntry entry = entries.get(0);
+
+    SimpleFeature feature = findDetailedFeatureForGeoEntry(entry);
+    if (feature != null) {
+      return feature;
+    }
+
+    GeoResult geoResult = GeoResultCreator.createGeoResult(entry);
+    return getFeatureFromGeoResult(geoResult);
```

Comment:
❓ Do we have any protection in here from returning bad features?  I'm wondering if there needs to be a validity check before returning (https://locationtech.github.io/jts/javadoc/org/locationtech/jts/geom/Geometry.html#isValid())

Reply:
The validation of bad features is done on the inbound (this is technically the outbound).  A bad feature wouldn't make it into the system.  https://github.com/codice/ddf/pull/2463/files#diff-4a2ff6ad9feb12e8f93ce66ae5c6fb1fR87